### PR TITLE
Add demo using fake finger driver & clean real finger demo

### PR DESCRIPTION
**Pull request by luator**
This is based on demo_real_finger.py but using the random finger driver
instead of the real one. Also I cleaned up a bit and added some explaining comments.

### CODE REVIEW

**Comment by luator** [comment]
I renamed `finger_backend` and also cleaned up and commented `demo_real_finger.py` (see second commit for details).  Both files are now identical except for the backend that is used.

**Comment by luator** [comment]
@mwuethri Can you please quickly re-review?

File: demos/demo_fake_finger.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def main():
+    # Storage for all observations, actions, etc.
+    finger_data = robot_interfaces.finger.Data()
+
+    # The backend sends actions from the data to the robot and writes
+    # observations from the robot to the data.  Here we use a backend using the
+    # "random finger driver" which just provides fake observations and does not
+    # need an actual robot to be executed.
+    fake_finger_backend = blmc_robots.create_random_finger_backend(finger_data)
+
```
Comment (mwuethri) [inline demos/demo_fake_finger.py]: on the right hand side you call it random_finger and on the left hand side you call it fake_finger.

my point was just that it should be consistent naming everywhere (including class names, file names, variable names, function names)

Comment (luator) [reply demos/demo_fake_finger.py]: I see. Given that the `RandomFingerDriver` does not really provide random observations (it is a sequence of increasing values) I would rather rename everything to "fake".  Going to do this in a separate PR, though.

**Pull request merged**